### PR TITLE
feat: add auth:web:login

### PR DESCRIPTION
**Pull request by mdonnalley**
### What does this PR do?
Adds the `auth:web:login` command

Depends on https://github.com/forcedotcom/sfdx-core/pull/313

### What issues does this PR fix or reference?
@W-7824743@

### CODE REVIEW

File: messages/web.login.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  "description": "authorizes a Salesforce org by opening a browser so you can log in through salesforce.com.",
+  "examples": [
+    "sfdx force:auth:web:login -a TestOrg1",
+    "sfdx force:auth:web:login -i <OAuth client id>",
+    "To log in to a sandbox, set --instanceurl to https://test.salesforce.com.",
+    "sfdx force:auth:web:login -r https://test.salesforce.com"
+  ],
+  "disableMasking": "disable masking of user input (for use with problematic terminals)",
+  "deviceWarning": "force:auth:web:login doesn't work when authorizing to a headless environment. Use force:auth:device:login instead.",
+  "clientSecretStdin": "OAuth client secret of personal connected app?",
```
Comment (rorolopetegui) [inline messages/web.login.json:11]: Please remove the ? symbol

File: yarn.lock [hunk 2]
```diff
@@ -401,7 +401,7 @@
     debug "^4.1.1"
     semver "^7.3.2"
 
-"@oclif/config@^1", "@oclif/config@^1.12.12", "@oclif/config@^1.15.1":
+"@oclif/config@^1.12.12", "@oclif/config@^1.15.1", "@oclif/config@^1.17.0":
```
Comment (rorolopetegui) [inline yarn.lock:404]: I have just a silly question here, are we pushing yarn.lock changes? What i mean, we're using this file for the CircleCi tests?

**Comment by rorolopetegui** [comment]
Can we fix this CircleCi Build problems, before merging?

**Comment by rorolopetegui** [comment]
> Can we fix this CircleCi Build problems, before merging?

I didnt see that this PR depends on the forcedotcom/sfdx-core#313, thats why the build is failing... sorry.

File: messages/web.login.json (hunk 1 above)
Comment (mdonnalley) [reply messages/web.login.json:11]: @rorolopetegui I'm porting these messages over from toolbelt without any modification: https://git.soma.salesforce.com/ALMSourceDrivenDev/force-com-toolbelt/blob/release/src/lib/messages.ts#L790

File: yarn.lock (hunk 2 above)
Comment (mdonnalley) [reply yarn.lock:404]: @rorolopetegui We include the yarn.lock to ensure dependencies are consistent across machines: https://classic.yarnpkg.com/en/docs/yarn-lock/#toc-check-into-source-control

So to answer, your question: yes, the yarn.lock is respected in the CircleCI tests

**Review by shetzel — APPROVED** [review]

**Review by amphro — APPROVED** [review]

File: test/commands/auth/device/login.test.ts [hunk 3]
```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/* eslint-disable @typescript-eslint/ban-ts-ignore */
```
Comment (WillieRuemmele) [inline test/commands/auth/device/login.test.ts:8]: If we're doing this in both of the test files, we should move it to the .eslintrc in the test dir

**Pull request merged**